src/ObjectUtils/PdbDiaUtils.cpp

```diff
@@ -0,0 +1,362 @@
+    IDiaSymbol* type, const std::string& parent_pointer_type_str) {
+  DWORD base_type;
+  if (FAILED(type->get_baseType(&base_type))) {
+    return ErrorMessage(orbit_base::GetLastErrorAsString());
+  }
+
+  std::string result_string;
+  switch (base_type) {
+    case btNoType:  // 0
+      result_string += "<no type>";
+      break;
+    case btVoid:  // 1
+      result_string += "void";
+      break;
+    case btChar:  // 2
+      result_string += "char";
+      break;
+    case btWChar:  // 3
+      result_string += "wchar_t";
+      break;
+    case btInt:  // 6
+    {
+      ULONGLONG length;
+      if (FAILED(type->get_length(&length))) {
+        return ErrorMessage(orbit_base::GetLastErrorAsString());
+      }
+      OUTCOME_TRY(auto&& integer_type, GetSignedIntegerTypeFromSizeInBytes(length));
+
+      result_string += integer_type;
+      break;
+    }
+    case btUInt:  // 7
+    {
+      ULONGLONG length;
+      if (FAILED(type->get_length(&length))) {
+        return ErrorMessage(orbit_base::GetLastErrorAsString());
+      }
+
+      result_string += "unsigned ";
+
+      OUTCOME_TRY(auto&& integer_type, GetSignedIntegerTypeFromSizeInBytes(length));
+
+      result_string += integer_type;
+      break;
+    }
+    case btFloat:  // 8
+    {
+      ULONGLONG length;
+      if (FAILED(type->get_length(&length))) {
+        return ErrorMessage(orbit_base::GetLastErrorAsString());
+      }
+
+      switch (length) {
+        case 4:
+          result_string += "float";
+          break;
+        case 8:
+          result_string += "double";
+          break;
+        default:
+          ORBIT_UNREACHABLE();
+      }
+      break;
+    }
+    case btBCD:  // 9
+      result_string += "<BCD>";
+      break;
+    case btBool:  // 10
+      result_string += "bool";
+      break;
+    case btLong:  // 13
+      result_string += "long";
+      break;
+    case btULong:  // 14
+      result_string += "unsigned long";
+      break;
+    case btCurrency:  // 25
+      result_string += "<currency>";
+      break;
+    case btDate:  // 26
+      result_string += "<date>";
+      break;
+    case btVariant:  // 27
+      result_string += "VARIANT";
+      break;
+    case btComplex:  // 28
+      result_string += "<complex>";
+      break;
+    case btBit:  // 29
+      result_string += "<bit>";
+      break;
+    case btBSTR:  // 30
+      result_string += "BSTR";
+      break;
+    case btHresult:  // 31
+      result_string += "HRESULT";
+      break;
+    case btChar16:  // 32
+      result_string += "char16_t";
+      break;
+    case btChar32:  // 33
+      result_string += "char32_t";
+      break;
+    case btChar8:  // 34
+      result_string += "char8_t";
+      break;
+    default:
+      return ErrorMessage(absl::StrFormat(
+          "Unexpected base type with id \"%d\". Returning \"???\" as a type name.", base_type));
+  }
+  result_string += parent_pointer_type_str;
+  return result_string;
```

Comment:
This function is still pretty long. May I suggest to move the handling of the parent_pointer_type_str out of this function. That would mean all the switch cases above could directly return instead of appending to a local string variable.

This would make it easier to follow for the reader and also shorter because the `break` statement won't be needed anymore. You could even shorten most of the cases to a single line: `case btChar8: return "char8_t"; // 34`

Reply:
Done. Also moved the extraction of the length in the integer case. This is no possible as we use errors.